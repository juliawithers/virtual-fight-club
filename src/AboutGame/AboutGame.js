import React, { Component } from 'react'
import context from '../context'

export default class AboutGame extends Component {
    static contextType = context;
    render(){
        return(
            <div>
                <header className="header" role="banner">
                    <h1>Virtual Fight Club</h1>
                </header>
                <section>
                    <h2>Character Development</h2>
                    <p>In Virtual Fight Club you can create a character and fight opponents! Get to the next level by gaining points from wins. 50 points to reach each new level. Once you reach the next level you can update the character attributes of your choosing. 10 attribute points are given at each level. Strength, Intelligence, Charisma, and Agility are the attributes you must work with. Each character is unique! </p>
                    
                    <p>And remember, the FIRST RULE OF FIGHT CLUB: You do not talk about (virtual) Fight Club!</p>
                </section>
                <section>
                    <h2>The FIGHT</h2>
                    <p>Each opponent is chosen at random and the winner is determined by an algorithm that incorporates the attribute values and random chance. Win and gain points towards your next upgrade. Lose and you won't lose anything, you just won't gain any points.</p>
                </section>
            </div>
        )
    }
}