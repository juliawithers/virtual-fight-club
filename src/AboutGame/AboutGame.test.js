import React from 'react';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom';
import AboutGame from './AboutGame'

import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe(`AboutGame component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <AboutGame/>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders the UI as expected',() =>{
        const wrapper = shallow(<AboutGame/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})