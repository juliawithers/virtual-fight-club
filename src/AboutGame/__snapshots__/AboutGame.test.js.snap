// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutGame component renders the UI as expected 1`] = `
<div>
  <header
    className="header"
    role="banner"
  >
    <h1>
      Virtual Fight Club
    </h1>
  </header>
  <section>
    <h2>
      Character Development
    </h2>
    <p>
      In Virtual Fight Club you can create a character and fight opponents! Get to the next level by gaining points from wins. 50 points to reach each new level. Once you reach the next level you can update the character attributes of your choosing. Strength, Intelligence, Charisma, and Agility are the attributes you must work with. Each character is unique! 
    </p>
    <p>
      And remember, the FIRST RULE OF FIGHT CLUB: You do not talk about (virtual) Fight Club!
    </p>
    <p>
      [placeholder for screenshot of character page]
    </p>
  </section>
  <section>
    <h2>
      The FIGHT
    </h2>
    <p>
      Each opponent is chosen at random and the winner is determined by an algorithm that incorporates the attribute values and random chance. Win and gain points towards your next upgrade. Lose and you won't lose anything, you just won't gain any points.
    </p>
    <p>
      [placeholder for screenshot of fight page]
    </p>
  </section>
</div>
`;
